### PR TITLE
Add follow-up fixes to Balena splash screen error.

### LINKED ISSUE #1648: Resolve issue with splash_screen
When running on Balena, the splash screen is crashing with the following error:

```
28.10.22 13:05:16 (+0000)  srly-ose-server  Exception on /splash-page [GET]
28.10.22 13:05:16 (+0000)  srly-ose-server  Traceback (most recent call last):
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2447, in wsgi_app
28.10.22 13:05:16 (+0000)  srly-ose-server      response = self.full_dispatch_request()
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1952, in full_dispatch_request
28.10.22 13:05:16 (+0000)  srly-ose-server      rv = self.handle_user_exception(e)
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/flask_restful/__init__.py", line 271, in error_router
28.10.22 13:05:16 (+0000)  srly-ose-server      return original_handler(e)
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/flask_cors/extension.py", line 165, in wrapped_function
28.10.22 13:05:16 (+0000)  srly-ose-server      return cors_after_request(app.make_response(f(*args, **kwargs)))
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1821, in handle_user_exception
28.10.22 13:05:16 (+0000)  srly-ose-server      reraise(exc_type, exc_value, tb)
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1950, in full_dispatch_request
28.10.22 13:05:16 (+0000)  srly-ose-server      rv = self.dispatch_request()
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1936, in dispatch_request
28.10.22 13:05:16 (+0000)  srly-ose-server      return self.view_functions[rule.endpoint](**req.view_args)
28.10.22 13:05:16 (+0000)  srly-ose-server    File "server.py", line 1812, in splash_page
28.10.22 13:05:16 (+0000)  srly-ose-server      my_ip = get_node_ip()
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/src/app/lib/utils.py", line 89, in get_node_ip
28.10.22 13:05:16 (+0000)  srly-ose-server      ), headers=headers)
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/requests/api.py", line 75, in get
28.10.22 13:05:16 (+0000)  srly-ose-server      return request('get', url, params=params, **kwargs)
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/requests/api.py", line 61, in request
28.10.22 13:05:16 (+0000)  srly-ose-server      return session.request(method=method, url=url, **kwargs)
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 529, in request
28.10.22 13:05:16 (+0000)  srly-ose-server      resp = self.send(prep, **send_kwargs)
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 645, in send
28.10.22 13:05:16 (+0000)  srly-ose-server      r = adapter.send(request, **kwargs)
28.10.22 13:05:16 (+0000)  srly-ose-server    File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 519, in send
28.10.22 13:05:16 (+0000)  srly-ose-server      raise ConnectionError(e, request=request)
28.10.22 13:05:16 (+0000)  srly-ose-server  ConnectionError: HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=e2446068f12647a691cb7a946151f3d4 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x720bd130>: Failed to establish a new connection: [Errno 111] Connection refused',))
```

Thee error appears to happen on both Pi 3 and Pi 4.

### PR DESCRIPTION
**Pull request by nicomiguelino**
Fixes #1648.

#### Errors

- The screen keeps displaying a page with the following **_Internal Server Error_** message &ndash; "The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application."

#### Logs

The latest logs come first.
- [screenly-ose-pi4_weathered-smoke-21.11.22_09_53_53_(-0800).txt](https://github.com/Screenly/screenly-ose/files/10059302/screenly-ose-pi4_weathered-smoke-21.11.22_09_53_53_.-0800.txt)
- [screenly-ose-pi4_weathered-smoke-21.11.22_09_27_00_(-0800).txt](https://github.com/Screenly/screenly-ose/files/10059298/screenly-ose-pi4_weathered-smoke-21.11.22_09_27_00_.-0800.txt)
- [screenly-ose-pi4_red-zone-07.11.22_07_12_00_(-0800).txt](https://github.com/Screenly/screenly-ose/files/9953963/screenly-ose-pi4_red-zone-07.11.22_07_12_00_.-0800.txt)

### CODE REVIEW

**Comment by nicomiguelino** [comment]
Hi, @vpetersson. Right now, I'm still getting the following error logs, even after the changes are applied.

```
07.11.22 07:02:52 (-0800) <srly-ose-server> HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=e9f16923fd1a475c8a6243e2aae328ba (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xf30f1f50>: Failed to establish a new connection: [Errno 111] Connection refused',)), retrying in 0 seconds...

07.11.22 07:02:52 (-0800) <srly-ose-server> HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=e9f16923fd1a475c8a6243e2aae328ba (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xf30f1c90>: Failed to establish a new connection: [Errno 111] Connection refused',)), retrying in 0 seconds...

07.11.22 07:02:52 (-0800) <srly-ose-server> HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=e9f16923fd1a475c8a6243e2aae328ba (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xf30f1450>: Failed to establish a new connection: [Errno 111] Connection refused',)), retrying in 0 seconds...

07.11.22 07:02:52 (-0800) <srly-ose-server> HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=e9f16923fd1a475c8a6243e2aae328ba (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xf30e6cf0>: Failed to establish a new connection: [Errno 111] Connection refused',)), retrying in 0 seconds...

07.11.22 07:02:52 (-0800) <srly-ose-server> HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=e9f16923fd1a475c8a6243e2aae328ba (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xf30c7bf0>: Failed to establish a new connection: [Errno 111] Connection refused',)), retrying in 0 seconds...
```

File: lib/utils.py [hunk 1]
```diff
@@ -83,7 +84,8 @@ def get_node_ip():
         requests.ConnectionError,
         requests.ConnectTimeout,
         requests.Timeout,
-    ), tries=3)
```
Comment (vpetersson) [inline lib/utils.py:86]: Did you intentionally lose the retry count?

Comment (nicomiguelino) [reply lib/utils.py:86]: Yes, I did it on purpose so that it'll keep retrying...

Comment (vpetersson) [reply lib/utils.py:86]: That's fair.

**Comment by nicomiguelino** [comment]
@vpetersson, here's what I'm getting with the latest changes applied.

```
...

21.11.22 10:53:55 (-0800) <srly-ose-server> HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=29f10dbf4c81474cb485191ecd04cbe2 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xf43a2af0>: Failed to establish a new connection: [Errno 111] Connection refused',)), retrying in 0 seconds...

21.11.22 10:53:55 (-0800) <srly-ose-server> HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=29f10dbf4c81474cb485191ecd04cbe2 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xf43a2310>: Failed to establish a new connection: [Errno 111] Connection refused',)), retrying in 0 seconds...

21.11.22 10:53:55 (-0800) <srly-ose-server> HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=29f10dbf4c81474cb485191ecd04cbe2 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xf43a25b0>: Failed to establish a new connection: [Errno 111] Connection refused',)), retrying in 0 seconds...

...
```

It just keeps spitting out that error message until the default Screenly page appears. This might be due to the removal of the `tries` argument.

Stay tuned for updates. I don't have any leads yet.

**Comment by nicomiguelino** [comment]
@vpetersson, with the most recent changes, **_502 Bad Gateway_** still appears on some occasions. I added a call to wait for the supervisor API to be ready on the **_viewer_** container as well. I'm still getting the following though...

```
22.11.22 19:30:39 (-0800) <srly-ose-server> HTTPConnectionPool(host='10.114.104.1', port=48484): Max retries exceeded with url: /v1/device?apikey=29f10dbf4c81474cb485191ecd04cbe2 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xf469b5d0>: Failed to establish a new connection: [Errno 111] Connection refused',)), retrying in 1 seconds...
```

[screenly-ose-pi4_weathered-smoke-22.11.22_19_50_27_(-0800).txt](https://github.com/Screenly/screenly-ose/files/10072703/screenly-ose-pi4_weathered-smoke-22.11.22_19_50_27_.-0800.txt)

File: lib/utils.py [hunk 2]
```diff
@@ -73,27 +73,15 @@ def validate_url(string):
...
-    return retry_call(
-        function,
-        delay=delay,
-        exceptions=(
-            requests.ConnectionError,
-            requests.ConnectTimeout,
-            requests.Timeout,
-            NewConnectionError,
-        ),
-    )
```
Comment (nicomiguelino) [inline lib/utils.py]: @vpetersson, I temporarily removed the usage of the `retry` library to test a hypothesis.

File: viewer.py [hunk 3]
```diff
@@ -442,7 +442,14 @@ def main():
...
 
-        get_supervisor_api_response(delay=1)
+        if is_balena_app():
+            retries = 5
+            for _ in range(retries):
+                try:
+                    get_supervisor_api_response()
+                    break
+                except requests.exceptions.ConnectionError:
+                    sleep(1)
```
Comment (nicomiguelino) [inline viewer.py:452]: @vpetersson, after powering off and then powering on the Pi 4 (not to be confused with rebooting it) five times, the Screenly splash page consistently shows up.

Given the current changes in this commit, the only instance where **_Bad Gateway_** occurs is when I deploy new changes on my device (via `balena push [fleet-name]`).

I'll change this block to use `retry` and test the changes a few times.

If the Splash page continues to show up consistently when powered on, #1648 can be closed. A separate issue can be created for instances when the Splash page does not show up in other circumstances (e.g., during reboot, container restart only, etc.). Asking a feedback for this one...

**Comment by nicomiguelino** [comment]
Just a sample comment done via GH CLI.

**Review by vpetersson — APPROVED** [review]

**Comment by nicomiguelino** [comment]
@ealmonte32, feel free to check this out for context

**Pull request merged**